low-level/actuators/impl/centos_7/systemd_service.py

```diff
@@ -80,44 +82,39 @@ def perform_request(self, jsonMsg):
             self._proxy = self._bus.get_object("org.freedesktop.systemd1", str(systemd_unit))
 
             # The returned result of the desired action
-            result = None
-
-            if self._service_request == "restart":
-                self._manager.RestartUnit(self._service_name, 'replace')
-
-                # Ensure we get an "active" status and not "activating"
-                max_wait = 0
-                while self._get_activestate() != "active":
-                    self._log_debug("status is still activating, pausing")
-                    time.sleep(1)
-                    max_wait += 1
-                    if max_wait > 20:
-                        self._log_debug("maximum wait for service restart reached")
-                        break
-
-            elif self._service_request == "start":
-                self._manager.StartUnit(self._service_name, 'replace')
-
-            elif self._service_request == "stop":
-                self._manager.StopUnit(self._service_name, 'replace')
+            result = {}
+            is_err_response = False
+            if self._service_request in ['restart', 'start', 'stop']:
+                Service('dbus').process(self._service_request, self._service_name)
+                if self._service_request == "restart":
+                    # Ensure we get an "active" status and not "activating"
+                    state, _, _, _  = Service('dbus').get_service_information(self._service_name)
+                    max_wait = 0
+                    while state != "active":
+                        logger.debug("status is still activating, pausing")
+                        time.sleep(1)
+                        max_wait += 1
+                        if max_wait > 20:
+                            logger.debug("maximum wait for service restart reached")

```

Comment:
This can be logger.error i feel, as this will result in error actuator response ?
logger.error("Max retries to restart %s service exhausted")

Reply:
No this will not give error actuator response, we will wait for some time and still if state != active, then in response we will send current service state and substate, whatever it is - failed, activating etc.

Reply:
added max_wait time in log statement.